scripts/ci-entrypoint.sh

```diff
@@ -305,9 +303,10 @@ on_exit() {
     unset KUBECONFIG
     go run -tags e2e "${REPO_ROOT}"/test/logger.go --name "${CLUSTER_NAME}" --namespace default
     "${REPO_ROOT}/hack/log/redact.sh" || true
-    # cleanup
-    if [[ -z "${SKIP_CLEANUP:-}" ]]; then
-        cleanup
+    # cleanup all resources we use
+    if [[ ! "${SKIP_CLEANUP:-}" == "true" ]]; then
+        "${KUBECTL}" --kubeconfig "${REPO_ROOT}/${KIND_CLUSTER_NAME}.kubeconfig" delete cluster "${CLUSTER_NAME}" --wait=true || echo "Unable to delete cluster ${CLUSTER_NAME} nodes"
```

Comment:
why remove the timeout here? I think that might cause the command to hang forever / until the job times out and then we wouldn't execute the rest nor print this error echo statement

Reply:
is `--wait true` actually doing anything? or is that just making it explicit

Reply:
Adding a timeout suggests that a successful operation lower than that threshold is an intrisic part of the test CI contract. A timeout combined with `|| true` is (IMO) the worst of both worlds, because we are enforcing a timeout but then ignoring its value.

The reason to remove the timeout would be to defer to the overall CI job timeout to enforce maximum long-running operation time.

Alternatively, we could pass `--wait=false`, and then add a sensible sleep statement (1 minute) to allow the capz controller to begin to set of delete operations w/ Azure before tearing down the mgmt cluster.

Reply:
IME `--wait=true` is the default, so yes just making this more explicit (and I could be wrong about that)

Reply:
I am still skeptical on this one, IME the kubectl delete --wait true function will not "fail" unless the delete webhook rejects the delete (for example the cluster we're trying to delete doesn't exist), what will actually happen here is the delete command will just hang forever if the delete is unsuccessful and it will never reach the `||` command.

The way I understand the current code is "try to wait this amount of time for the cluster to be deleted but move on if that is unsuccessful". 

> The reason to remove the timeout would be to defer to the overall CI job timeout to enforce maximum long-running operation time.

I don't see this as beneficial because then we wouldn't give the job a chance to print out any useful information about why the delete failed and perform any follow-up steps like deleting the kind cluster, and the delete would take potentially up to 4 hours (the timeout we set for most conformance jobs) as opposed to the current 30 minutes. We expect a conformance test suite to take up to 4 hours, so it's an appropriate timeout, but a delete that takes more than 30 minutes is very likely to never succeed, so there's little use in hogging compute resources for those extra 3.5 hours.

Reply:
Is the delete command essentially waiting for a very specific status phase transition in the `Cluster` resource? If there was a terminal issue preventing the delete command from completing we would see that in the logs, no? And we should also get a failed test run if the overall timeout allotted is exceeded. I think that's the desired outcome.

Having a concrete timeout around the delete operation would also give us the desired non-zero test result (and thus encourage us to investigate what happened in the logs), do we want to carry over the 30 minute timeout (via `test/e2e/config/azure-dev.yaml`) that we currently require for a fully passing test outcome on CAPZ E2E for ci-entrypoint-dependent test jobs?

Reply:
> Is the delete command essentially waiting for a very specific status phase transition in the Cluster resource?

not exactly, it's waiting for the object to no longer exist, after k8s has garbage collected the object (which happens once the CAPI controller removes the finalizer).

> And we should also get a failed test run if the overall timeout allotted is exceeded. I think that's the desired outcome.

If the desired outcome is to fail the run on delete, then we could just remove the `|| true` part. We don't need to wait 4 hours to determine whether the delete really failed. I think there was a good reason for not doing it this way at the time when this was implemented, which is that this script is _not_ meant to test CAPZ itself, but rather a tool for creating test clusters to run other tests such as cloud-provider e2e, so we didn't want to fail the test run if the thing we're testing (e.g. the cloud-provider test suite ran and all tests passed) was successful to not slow down/block development of other projects.

Reply:
one argument for _not_ doing this right now: we aren't confident that delete is working and has ever worked for CCM jobs so if we merge this with the "fail on delete failure" change and it turns out delete is not working (and this doesn't fix it) we could cause all k8s / CCM jobs to be broken until we figure it out. Maybe a good compromise for now is to just log the failure, then once we're confident delete is working as expected, we switch to a strict fail.

Reply:
I think that makes sense. Any objections to increasing the timeout to 30m to match CAPZ E2E, or should we just stick to 20m to decrease the chances that we'll stumble onto something new and regress downstream tests further?

Reply:
the existing timeout of 1800s is 30 minutes already no?

Reply:
Dog ate my calculator

Reply:
Restored previous implementation w/ echo statement replacing `true` assertion in the `||` path